File: ios/Config/RNConfig.m
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wnullability-completeness"

#if __has_include("RCTBridgeModule.h")
#import "RCTBridgeModule.h"
#else
#import <React/RCTBridgeModule.h>
#endif

@interface RCT_EXTERN_MODULE(RNConfig, NSObject)

RCT_EXTERN_METHOD(getAppVersion:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject);
RCT_EXTERN_METHOD(getDeviceId:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject);
RCT_EXTERN_METHOD(exitApp:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject);

@end

#pragma clang diagnostic pop

File: ios/Config/RNConfig.swift
import Foundation

@objc(RNConfig)
@objcMembers final class RNConfig: NSObject {
  @objc(getAppVersion:reject:)
  func getAppVersion(resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    let appVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"] ?? "0.0.1"
    resolve(appVersion)
  }

  @objc(getDeviceId:reject:)
  func getDeviceId(resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    let deviceId = UIDevice.current.identifierForVendor!.uuidString
    resolve(deviceId)
  }

  @objc(exitApp:reject:)
  func exitApp(resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    exit(0)
  }
}
